### PR TITLE
Add test for model requirements

### PR DESCRIPTION
**Pull request by danielholanda**
Closes https://github.com/groq/mlagility/issues/106
This PR adds tests to ensure that we can successfully install MLAgility along all requirements of `requirements.txt`.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -28,9 +28,22 @@ jobs:
...
           if [ -f setup.py ]; then pip install -e .; fi
           pip install transformers
+
+          #Install miniconda
+          wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+          bash Miniconda3-latest-Linux-x86_64.sh -b -u
       - name: Lint with PyLint
         run: |
           pylint src --rcfile .pylintrc
+      - name: Check installation of model requirements on isolated conda env
```
Comment (jeremyfowers) [inline .github/workflows/test.yml:38]: Move after unittest, since this is not a unit test

**Review by jeremyfowers — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (danielholanda) [reply .github/workflows/test.yml:42]: done

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -28,9 +28,22 @@ jobs:
...
+        run: |
+          python test/check_install.py models/requirements.txt --expect_failure
+          source ~/miniconda3/bin/activate
+          bash src/mlagility/cli/setup_venv.sh $PWD test_env
+          conda activate test_env
+          python test/check_install.py models/requirements.txt
```
[the remark this answers is not in the document]
Comment (danielholanda) [reply .github/workflows/test.yml:48]: done

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -28,10 +28,24 @@ jobs:
           pip install pylint
           if [ -f setup.py ]; then pip install -e .; fi
           pip install transformers
+
+          #Install miniconda
+          wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+          bash Miniconda3-latest-Linux-x86_64.sh -b -u
```
Comment (jeremyfowers) [inline .github/workflows/test.yml:34]: fyi I'm not sure how to make this work with localhost benchmarking. I tried and got weird errors. Can you wait for #104 to merge and then see if your stuff still works?

**Review by jeremyfowers — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 3 above)
Comment (danielholanda) [reply .github/workflows/test.yml:34]: This works well locally as long as `source ~/miniconda3/bin/activate` is used

Comment (jeremyfowers) [reply .github/workflows/test.yml:34]: I tried that but got hit with
```
| ERROR: Package 'mlagility' requires a different Python: 3.10.9 not in '==3.8.*'
```

**Review by jeremyfowers — APPROVED** [review]

**Pull request merged**